contracts/math/uints.ral

```diff
@@ -4,18 +4,74 @@ struct U512T{
 }
 
 Abstract Contract U512() {
-    pub fn bigAdd(a: U256, b: U256) -> U512T {
-        let u256Max = u256Max!()
-        if (u256Max - a < b) {
+        pub fn bigAdd(a: U256, b: U256) -> U512T {
+            let u256Max = u256Max!()
+            if (u256Max - a < b) {
+                return U512T {
+                    higher: 1,
+                    lower: (b - (u256Max - a)) - 1
+                }
+            } else {
+                return U512T {
+                    higher: 0,
+                    lower: a + b
+                }
+            }
+        }
+
+        pub fn bigDiv(a: U512T, divisor: U256, divisorScale: U256) -> U512T {
+            assert!(divisor > 0, 0x02)
+            if (divisor == 1) {
+                return a
+            }
+
+
+            let max = u256Max!()
+            let scale = 10 ** divisorScale
+            // Calculate new higher
+            let h = (a.higher * scale) / divisor
```

Comment:
Try to optimize this to reduce the amount of calculation. GPT will be helpful for this exercise. For example, you calculate `a.higher * scale` twice. Also, check if this formula itself can be simplified